packages/fakta/omsorg-og-rett/src/components/forms/RettighetstypeForm.tsx

```diff
@@ -0,0 +1,121 @@
+  rettighetstype: Rettighetstype;
+  begrunnelse: string;
+};
+
+interface Props {
+  omsorgOgRett: OmsorgOgRett;
+  aksjonspunkt?: Aksjonspunkt;
+  kanOverstyre: boolean;
+  submittable: boolean;
+}
+
+const RETTIGHETSTYPER = {
+  [Rettighetstype.ALENEOMSORG]: 'Rettighetstype.Aleneomsorg',
+  [Rettighetstype.BEGGE_RETT]: 'Rettighetstype.BeggeRett',
+  [Rettighetstype.BEGGE_RETT_EØS]: 'Rettighetstype.BeggeRettEØS',
+  [Rettighetstype.BARE_MOR_RETT]: 'Rettighetstype.BareMorRett',
+  [Rettighetstype.BARE_FAR_RETT]: 'Rettighetstype.BareFarRett',
+  [Rettighetstype.BARE_FAR_RETT_MOR_UFØR]: 'Rettighetstype.BareFarRettMorUfør',
+} as Record<string, string>;
+
+export const RettighetstypeForm = ({ omsorgOgRett, aksjonspunkt, submittable, kanOverstyre }: Props) => {
+  const { submitCallback, alleMerknaderFraBeslutter, isReadOnly } = usePanelDataContext<OverstyringRettigheterAp>();
+  const intl = useIntl();
+
+  const rettighetstype = omsorgOgRett.rettighetstype ?? undefined;
+
+  const { mellomlagretFormData, setMellomlagretFormData } = useMellomlagretFormData<FormValues>();
+
+  const formMethods = useForm<FormValues>({
+    defaultValues: mellomlagretFormData ?? {
+      rettighetstype: rettighetstype,
+      ...FaktaBegrunnelseTextField.initialValues(aksjonspunkt),
+    },
+  });
+
+  const transformerFeltverdier = useCallback(
+    (feltVerdier: FormValues) =>
+      submitCallback({
+        kode: AksjonspunktKode.OVERSTYRING_AV_RETT_OG_OMSORG,
+        rettighetstype: feltVerdier.rettighetstype,
+        ...FaktaBegrunnelseTextField.transformValues(feltVerdier),
+      }),
+    [],
+  );
+  const rettighetstyper =
+    omsorgOgRett.relasjonsRolleType === RelasjonsRolleType.MOR
+      ? [
+          Rettighetstype.ALENEOMSORG,
+          Rettighetstype.BEGGE_RETT,
+          Rettighetstype.BEGGE_RETT_EØS,
+          Rettighetstype.BARE_MOR_RETT,
+        ]
+      : [
+          Rettighetstype.ALENEOMSORG,
+          Rettighetstype.BEGGE_RETT,
+          Rettighetstype.BEGGE_RETT_EØS,
+          Rettighetstype.BARE_FAR_RETT,
+          Rettighetstype.BARE_FAR_RETT_MOR_UFØR,
+        ];
+  const [erOverstyrt, setErOverstyrt] = useState(!!aksjonspunkt?.begrunnelse);
+  const readOnly = !erOverstyrt || isReadOnly || !kanOverstyre;
+  return (
+    <Form formMethods={formMethods} onSubmit={transformerFeltverdier} setDataOnUnmount={setMellomlagretFormData}>
+      <FaktaGruppe
+        withoutBorder
+        merknaderFraBeslutter={alleMerknaderFraBeslutter[AksjonspunktKode.OVERSTYRING_AV_RETT_OG_OMSORG]}
+      >
+        <VStack gap="6">
+          <HStack gap="2" align="start">
+            <SelectField
+              name="rettighetstype"
+              label="Rettighetstype"
+              validate={[required]}
+              className={styles.selectType}
+              selectValues={rettighetstyper.map(vt => (
+                <option key={vt} value={vt}>
+                  {intl.formatMessage({ id: RETTIGHETSTYPER[vt] })}
```

Comment:
Her kan du med fordel bruka komponent i staden for. Altso: <FormattedMessage ... />. Så blir det litt meir lesbart